internal/namespaces/instance/v1/custom_image.go

```diff
@@ -266,21 +269,49 @@ func imageDeleteBuilder(c *core.Command) *core.Command {
 			return nil, err
 		}
 
+		type UnknownSnapshot struct {
+			ID   string
+			Type instance.VolumeVolumeType
+		}
+
 		// Once the image is deleted we can delete snapshots.
 		if args.WithSnapshots {
-			snapshotIDs := []string{
-				image.RootVolume.ID,
+			snapshots := []UnknownSnapshot{
+				{
+					ID:   image.RootVolume.ID,
+					Type: image.RootVolume.VolumeType,
+				},
 			}
-			for _, snapshot := range image.ExtraVolumes {
-				snapshotIDs = append(snapshotIDs, snapshot.ID)
-			}
-			for _, snapshotID := range snapshotIDs {
-				err := api.DeleteSnapshot(&instance.DeleteSnapshotRequest{
-					Zone:       args.Zone,
-					SnapshotID: snapshotID,
+			for _, extraVolume := range image.ExtraVolumes {
+				snapshots = append(snapshots, UnknownSnapshot{
+					ID:   extraVolume.ID,
+					Type: extraVolume.VolumeType,
 				})
-				if err != nil {
-					return nil, err
+			}
+			for _, snapshot := range snapshots {
+				if strings.HasPrefix(string(snapshot.Type), "sbs") {
```

Comment:
`sbs` seems inconsistant with block namespace which is used in the sdk. Is it possible to use `block`? It seems easier to remember and complete.

Reply:
I solved it using instance.VolumeVolumeType enum